### ISSUE TITLE
mlemodel.PredictionResults.summary_frame attached self.table... why?

### ISSUE THREAD

**Issue by jbrockmendel** [statsmodels/statsmodels#5271]
@ChadFulton doesn't appear used, isn't a sm convention, and makes me wince because it isn't in `__init__`.  What's the reasoning for this?

**Comment by ChadFulton** [comment]
No idea - looks like it came in the initial scikits commit.

(It often comes up first in Sublime Text when I'm actually trying to go to statespace/mlemodel, so I'm all for getting rid of it if it's not used :)

**Comment by josef-pkt** [comment]
Where is this?
I remember we did this in the past to have access to the underlying dataframe (similar to a cached attribute), but I didn't find it anymore when I went briefly through some summary methods.

**Comment by josef-pkt** [comment]
I found it. I used it in both genmod._prediction and in regression._prediction
I don't see a reason given in my original PR.
There is no caching, no use of cached attributes in PredictResults. So, I guess I wanted to cache the results at least somewhere to avoid recomputing.

design for get_prediction was preliminary (i.e. without thinking to hard about all the details), and never got a "finalizing" round.
